packages/safety-suite-api/src/api/models/types.ts

```diff
@@ -115,23 +115,281 @@ export type OrderBy = {
 };
 
 export type ModelListOptions = {
+  /**
+   * Maximum number of models to return.
+   */
   limit?: number;
+  /**
+   * Options for sorting the returned models.
+   */
   orderBy?: OrderBy;
+  /**
+   * Index in the array of returned models.
+   */
   start?: number;
 };
 
 export type ModelOutputOptions = {
+  /**
+   * Will include `profilePictureUrl` in returned models if present.
+   */
   profilePicture?: boolean;
+  /**
+   * Will include confidential fields in returned models if permitted.
+   */
   protectedFields?: boolean;
+  /**
+   * Will include related models in returned models.
+   * `singleModel` will return only models with a singleModel relation.
+   * `all` will return all types of relations.
+   *
+   * Depth is set by `relationsLevel`.
+   */
   relationModels?: string;
   relationNames?: string[];
   relationTemplateIds?: Id[];
+  /**
+   * Depth of related models to retrieve.
+   *
+   * Example:
+   * ```ts
+   * // Output model with relationsLevel === 2
+   * {
+   *   // ...
+   *   relationModels: {
+   *     accident: {
+   *       // ...
+   *       relationModels: {
+   *         employee: {
+   *           // ...
+   *         }
+   *       }
+   *     }
+   *   }
+   * }
+   * ```
+   */
   relationsLevel?: number;
+  /**
+   * Array of names of computations with `computeOn === 'read'` to skip computation.
+   */
   restrictReadComputations?: string[];
+  /**
+   * Array of names of computations with `computeOn === 'read'` to force computation.
+   */
   updateReadComputations?: boolean;
+  /**
+   * Will include `createdByUser` in returned models.
+   */
   users?: boolean;
 };
 
 export type ModelOptions = ModelFilterOptions &
   ModelListOptions &
   ModelOutputOptions;
+
+/**
+ * Determines how `queryPath` will be used.
+ */
+export enum QueryPathType {
+  /**
+   * Used to match relation ids.
+   *
+   * Example:
+   * ```ts
+   * operation: {type: OperationType.equals},
+   * queryPath: {
+   *   pathType: QueryPathType.relation,
+   *   path: ['employee']
+   * },
+   * queryValue: {value: 1} // ID of an Employee model
+   * ```
+   */
+  relation = 'RELATION',
+  /**
+   * Used to match field or computation values.
+   *
+   * Example:
+   * ```ts
+   * queryPath: {
+   *   pathType: QueryPathType.path,
+   *   path: ['firstName']
+   * }
+   * ```
+   */
+  path = 'PATH',
+  group = 'GROUP',
+  template = 'TEMPLATE',
```

Comment:
Might be worth mentioning that these options only match group and template ids, they will not work for aliases or smth like that.